addon/normalizer.js

```diff
@@ -38,39 +38,48 @@ export default class Normalizer {
     const type = this.primaryModelClass.modelName;
     const isCollection = false;
 
-    this._extractSingle(resourcePayload, type, isCollection);
+    this._extractSingle(resourcePayload, type, type, isCollection);
     return this.response.toJSON();
   }
 
-  _extractCollection(payloads, type) {
+  _extractCollection(payloads, key, type) {
     const isCollection = true;
-    payloads.forEach((payload) => this._extractSingle(payload, type, isCollection));
+    payloads.forEach((payload) => this._extractSingle(payload, key, type, isCollection));
   }
 
-  _extractSingle(payload, type, isCollection=false) {
+  _extractSingle(payload, key, type, isCollection=false) {
     const attributes = copyAttributesExcluding(payload, RESERVED_ATTRIBUTES);
     const id = payload.id;
-    const resource = new Resource({ id, type, attributes });
+    const resource = new Resource({ id, key, type, attributes });
 
     this.response.pushResource(resource, isCollection, () => {
-      this._extractRelationships(payload._embedded || {});
+      var primaryModelClass = this.primaryModelClass;
+      this._extractRelationships(payload._embedded || {}, primaryModelClass);
     });
   }
 
-  _extractRelationships(payload) {
+  _extractRelationships(payload, primaryModelClass) {
     Object.keys(payload).forEach((key) => {
       const relatedPayload = payload[key];
-      const type = key;
-      this._extractRelationship(relatedPayload, type);
+      const attributeName = Ember.String.camelize(key);
```

Comment:
This bakes in an assumption about how people name their relationships. Someone should be able to:
```
DS.Model.extend({
});
```